src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -0,0 +1,122 @@
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import withAdminPolicyAccessOrNotFound from '@pages/workspace/withAdminPolicyAccessOrNotFound';
+import type {WithAdminPolicyAccessOrNotFoundProps} from '@pages/workspace/withAdminPolicyAccessOrNotFound';
+import withPaidPolicyAccessOrNotFound from '@pages/workspace/withPaidPolicyAccessOrNotFound';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+
+type PolicyForList = {
+    value: string;
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+    rightElement: React.ReactNode;
+};
+
+type WorkspaceCategoriesOnyxProps = {
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<OnyxTypes.PolicyCategories>;
+};
+
+type WorkspaceCategoriesPageProps = WorkspaceCategoriesOnyxProps &
+    WithAdminPolicyAccessOrNotFoundProps &
+    StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORIES>;
+
+function WorkspaceCategoriesPage({policyCategories}: WorkspaceCategoriesPageProps) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+    const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+
+    const categoryList = useMemo<PolicyForList[]>(
+        () =>
+            Object.values(policyCategories ?? {}).map((value) => ({
+                value: value.name,
+                text: value.name,
+                keyForList: value.name,
+                isSelected: selectedCategories.includes(value.name),
+                rightElement: (
+                    <View style={styles.flexRow}>
+                        <Text style={[styles.disabledText, styles.alignSelfCenter]}>{value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}</Text>
+                        <View style={[styles.p1, styles.pl2]}>
+                            <Icon
+                                src={Expensicons.ArrowRight}
+                                fill={theme.icon}
+                            />
+                        </View>
+                    </View>
+                ),
+            })),
+        [policyCategories, selectedCategories, styles.alignSelfCenter, styles.disabledText, styles.flexRow, styles.p1, styles.pl2, theme.icon, translate],
+    );
+
+    const toggleCategory = (category: PolicyForList) => {
+        setSelectedCategories((prev) => {
+            if (prev.includes(category.value)) {
+                return prev.filter((item) => item !== category.value);
+            }
+            return [...prev, category.value];
+        });
```

Comment:
Same here, using an object might make this easier

Reply:
done!